numpy/random/_generator.pyx

```diff
@@ -4162,7 +4216,127 @@ cdef class Generator:
 
         return diric
 
-    # Shuffling and permutations:
+    def randomly_permute(self, object x, axis=None):
+        """
+        Randomly permute `x` along axis `axis` in-place.
+
+        If `axis` is None, the flattened array is shuffled.
+
+        Unlike `shuffle`, each slice along the given axis is shuffled
+        independently of the others.
+
+        Parameters
+        ----------
+        x : ndarray
+            Array to be randomly permuted.
+        axis : int, optional
+            Slices of `x` in this axis are permuted. Each slice
+            is permuted independently of the others.
+
+        Returns
+        -------
+        None
+
+        See Also
+        --------
+        numpy.random.Generator.randomly_permuted
+        numpy.random.Generator.shuffle
+        numpy.random.Generator.permutation
+
+        Examples
+        --------
+        Create a `numpy.random.Generator` instance:
+
+        >>> rng = np.random.Generator(np.random.PCG64(12345))
```

Comment:
default_rng() is preferred I think

Reply:
OK.  I'm not sure if the display of the random samples shown below will then break some doctests. I can add the appropriate markup/work-around if it does.